src/pages/trails/treasuries/components/TreasuryTokenSettingsEdit.vue

```diff
@@ -0,0 +1,77 @@
+<script>
+import { validation } from '~/mixins/validation';
+
+export default {
+  name: 'TreasuryTokenSettingsEdit',
+  mixins: [validation],
+  props: {
+    modelValue: {
+        type: Array,
+        required: true
+    },
+  },
+  data() {
+    return {
+      preValue: {},
+      selected: {
+        key: '',
+        body: ''
+      }
+    };
+  },
+  async mounted() {
+    this.remember();
+  },
+  methods: {
+    remember(newValue) {
+      this.preValue = (newValue || this.modelValue).map(x => ({...x}))
+    }
+  },
+  watch: {
+    modelValue: {
+      handler:function(newval) {
+        let changes = this.preValue.filter((x,i) => {          
+          let result = newval[i].value != x.value;
+          return result;
+        });
+
+        if (changes.length == 1) {
+          this.selected = {
+            key: changes[0].key,
+            body: 'pages.trails.treasuries.settings.' + changes[0].key
+          }
+        }
+        this.remember();
+      },
+      deep:true
+    }
+  }
+};
+</script>
+
+<template lang="pug">
```

Comment:
Another opportunity for not using pug

Reply:
done here [fcec55c](https://github.com/telosnetwork/app-telos-native/pull/224/commits/fcec55c351ecae17f26547bf79fc6c7f0e6d855c)